test/distributed/optim/test_zero_redundancy_optimizer.py

```diff
@@ -204,6 +203,40 @@ def test_zero_grad(self):
         self.assertFalse(m.weight.grad)
         self.assertFalse(m.bias.grad)
 
+    def test_constructor(self):
+        """Check the robustness of the ZeroRedundancyOptimizer constructor by
+        passing different values for `params`"""
+        self.dist_init(self.rank)
+
+        # Non-iterable: `torch.Tensor`
+        with self.assertRaises(TypeError):
+            ZeroRedundancyOptimizer(torch.randn(1), optimizer_class=SGD, lr=0.1)
+
+        # Non-iterable: `float`
+        with self.assertRaises(TypeError):
+            ZeroRedundancyOptimizer(1.2, optimizer_class=SGD, lr=0.1)
+
+        # Empty parameter list
+        with self.assertRaises(ValueError):
+            ZeroRedundancyOptimizer([], optimizer_class=SGD, lr=0.1)
+
+        # Iterable of dict
+        m = torch.nn.Linear(1, 1)
+        params = [{"params": m.parameters()}]
+        with self.assertRaises(TypeError):
+            ZeroRedundancyOptimizer(params, optimizer_class=SGD, lr=0.1)
+
+        # Iterable containing non-`torch.Tensor`
+        params = list(m.parameters()) + [42]
+        with self.assertRaises(TypeError):
+            ZeroRedundancyOptimizer(params, optimizer_class=SGD, lr=0.1)
```

Comment:
nit: since all these are checking for the same error, you can just construct a list of invalid inputs and assert in a loop to reduce code duplication?

Reply:
Fixed to loop over inputs.